### PR TITLE
feat(chrome-scans): Add `ChromiumComponentsShouldUseWebScanner` rule

### PR DESCRIPTION
**Pull request by sfoslund**
#### Details

This PR is 1/2 of addressing https://github.com/microsoft/axe-windows/issues/836. It includes the following commits/changes, which can be reviewed separately:

- https://github.com/microsoft/axe-windows/commit/27295d6a0c22942f26d72efe62be886ce0da5460 adds a new sample project to the tools repo that uses a minimal webview2.
- https://github.com/microsoft/axe-windows/commit/b0076947294e73f061ac56ce349979296a57a8d7 adds the `ChromiumComponentsShouldUseWebScanner` rule and a new e2e test leveraging the new web view sample added in this PR. 

##### Motivation

Addresses issue https://github.com/microsoft/axe-windows/issues/836

##### Context

This PR only adds the new rule logic but does not do the work to suppress any further errors in descendants of the root "Chrome" element (hence why there are 35 expected errors in the end to end test instead of just 1). A future PR will add the suppression logic separately. 

#### Pull request checklist

- [x] Addresses an existing issue: [#0000](https://github.com/microsoft/axe-windows/issues/836)

### CODE REVIEW

File: src/AutomationTests/AutomationIntegrationTests.cs [hunk 1]
```diff
@@ -26,12 +26,14 @@ public class AutomationIntegrationTests
         const int WindowsFormsControlSamplerKnownErrorCount = 6;
         const int WpfControlSamplerKnownErrorCount = 7;
         const int WindowsFormsMultiWindowSamplerAppAllErrorCount = 12;
+        const int WebViewSampleKnownErrorCount = 35;
```
Comment (codeofdusk) [inline src/AutomationTests/AutomationIntegrationTests.cs:29]: Maybe add a note that this will eventually be reduced to 1 error?

**Review by codeofdusk — APPROVED** [review]
lgtm

File: src/Rules/Resources/Descriptions.resx [hunk 2]
```diff
@@ -520,4 +520,8 @@ HowToFix:    {2}
 Condition:  {3}</value>
     <comment>{0} is id, {1} is description, {2} is how to fix, and {3} is condition</comment>
   </data>
+  <data name="ChromiumComponentsShouldUseWebScanner" xml:space="preserve">
+    <value>Chromium components should be scanned with a web-based scanner.</value>
```
[the remark this answers is not in the document]
Comment (codeofdusk) [reply src/Rules/Resources/Descriptions.resx:524]: The "how to fix" guidance suggests K4W.

File: src/Rules/Library/ChromiumComponentsShouldUseWebScanner.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+namespace Axe.Windows.Rules.Library
+{
+    [RuleInfo(ID = RuleId.ChromiumComponentsShouldUseWebScanner)]
+    class ChromiumComponentsShouldUseWebScanner : Rule
+    {
+        public ChromiumComponentsShouldUseWebScanner()
+        {
+            Info.Description = Descriptions.ChromiumComponentsShouldUseWebScanner;
+            Info.HowToFix = HowToFix.ChromiumComponentsShouldUseWebScanner;
+            Info.ErrorCode = EvaluationCode.Error;
```
Comment (DaveTryon) [inline src/Rules/Library/ChromiumComponentsShouldUseWebScanner.cs:18]: At one point, someone mentioned treating this as a framework issue, which would potentially mean creating a web page and linking to it. Did we ultimately decide against that approach?

Comment (sfoslund) [reply src/Rules/Library/ChromiumComponentsShouldUseWebScanner.cs:18]: I wasn't aware of that discussion and Fer is OOF so we won't be able to get her input for a while. Is that something that had been settled previously?

File: src/RulesTest/Library/EdgeBrowserHasBeenDeprecatedTests.cs [hunk 4]
```diff
@@ -29,7 +29,7 @@ public void FrameworkIssueLink_IsNotNull()
         [TestMethod]
         public void Condition_FrameworkIsNotEdge_ReturnsFalse()
         {
-            string[] nonWin32Values = { FrameworkId.DirectUI, FrameworkId.InternetExplorer, FrameworkId.WinForm, FrameworkId.WPF, FrameworkId.XAML, FrameworkId.Win32, "NotEdge" };
+            string[] nonWin32Values = { FrameworkId.DirectUI, FrameworkId.InternetExplorer, FrameworkId.WinForm, FrameworkId.WPF, FrameworkId.XAML, FrameworkId.Win32, FrameworkId.Chrome, "NotEdge" };
```
Comment (DaveTryon) [inline src/RulesTest/Library/EdgeBrowserHasBeenDeprecatedTests.cs:32]: I find 3 other test classes where we take a similar approach:
* ControlShouldSupportSetInfoWPFTests.cs
* ControlShouldSupportSetInfoXAMLTests.cs
* FrameworkDoesNotSupportUIAutomationTests.cs

Perhaps a future PR could add a utility method that would have a signature along the lines of
```
IEnumerable<string> GetFilteredFrameworks(string excludeThis, string addThis);
```
and we could use that method in all of them. Assuming that method were to use reflection to enumerate the values, adding a fields added to the `FrameworkId` class would automatically get picked up in the test classes.

File: tools/WebViewSample/src/WebViewSample.csproj [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>WinExe</OutputType>
+    <TargetFramework>net7.0-windows</TargetFramework>
```
Comment (DaveTryon) [inline tools/WebViewSample/src/WebViewSample.csproj:5]: We're using net6.0 in all other tests. For consistency, should we also use it here? According to https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core, net6.0 is LTS, while net7.0 is not.

File: src/RulesTest/Library/EdgeBrowserHasBeenDeprecatedTests.cs (hunk 4 above)
Comment (sfoslund) [reply src/RulesTest/Library/EdgeBrowserHasBeenDeprecatedTests.cs:32]: Good idea! Created a separate PR to add that utility: https://github.com/microsoft/axe-windows/pull/872

**Review by DaveTryon — APPROVED** [review]

**Pull request merged**